### PR TITLE
further prune column for TachyonTableRDD

### PR DESCRIPTION
**Pull request by haoyuan**

### CODE REVIEW

**Comment by rxin** [comment]
Jenkins, retest this please.

File: src/tachyon_enabled/scala/shark/tachyon/TachyonTableRDD.scala [hunk 1]
```diff
@@ -58,36 +64,44 @@ class TachyonTableRDD(path: String, @transient sc: SparkContext)
...
-      if (buf == null) {
-        logWarning("Table " + path + " column " + columnIndex + " partition " + theSplit.index
-          + " is not in Tachyon's memory. Streaming it in.")
-        var data = ByteBuffer.allocate(fp.length().toInt)
-        val is = fp.getInStream(ReadType.CACHE)
-        is.read(data.array)
-        is.close()
-        data.limit(fp.length().toInt)
-        data
+      if (columnIndex != 0 && mColumnUsed != null && !mColumnUsed.get(columnIndex - 1)) {
```
Comment (rxin) [inline src/tachyon_enabled/scala/shark/tachyon/TachyonTableRDD.scala:70]: Actually let's change TablePartition to skip null buffers, and then we don't need to initialize these dummy buffers here.

File: src/main/scala/shark/execution/TableScanOperator.scala [hunk 2]
```diff
@@ -148,6 +148,9 @@ class TableScanOperator extends TopOperator[HiveTableScanOperator] with HiveTopO
 
     // Run map pruning if the flag is set, there exists a filter predicate on
     // the input table and we have statistics on the table.
+    var columnsUsed = new ColumnPruner(this, table).columnsUsed;
```
Comment (rxin) [inline src/main/scala/shark/execution/TableScanOperator.scala]: var -> val, and remove the trailing ";"

File: src/tachyon_enabled/scala/shark/tachyon/TachyonUtilImpl.scala [hunk 3]
```diff
@@ -42,6 +43,16 @@ class TachyonUtilImpl(val master: String, val warehousePath: String) extends Tac
 
   def getPath(tableName: String): String = warehousePath + "/" + tableName
 
+  override def addColumnPruneInfo(rdd: RDD[_], columnUsed: BitSet): Boolean = {
```
Comment (rxin) [inline src/tachyon_enabled/scala/shark/tachyon/TachyonUtilImpl.scala]: Rename this to pushDownColumnPruning so it is more database-y :)

File: src/tachyon_enabled/scala/shark/tachyon/TachyonTableRDD.scala (hunk 1 above)
Comment (haoyuan) [inline src/tachyon_enabled/scala/shark/tachyon/TachyonTableRDD.scala:70]: done

**Pull request merged**